src/components/pets/PetInfo.js

```diff
@@ -15,22 +15,40 @@ import { GiAges } from "react-icons/gi";
 import { HiMail } from "react-icons/hi";
 import "./PetInfo.css";
 import { usePetAuth } from "../../context/TokenContext";
-import useFetch from "../../useHooks/useFetch";
+// import useFetch from "../../useHooks/useFetch";
 import { lookUpPet } from "../../routes/API";
 import LoaderComponent from "../../utils/LoaderComponent";
+import useSWR from "swr";
+import axios from "axios";
 
 export default function PetInfo() {
   const { id } = useParams();
+  const [isLoading, setIsLoading] = useState(false);
+  const fetcher = (url, tokenHeader) =>
+    axios
+      .get(url, {
+        headers: tokenHeader.headers,
+      })
+      .then((res) => {
+        res(data);
+      });
 
   const { tokenHeaders } = usePetAuth();
 
-  const { data, isLoading, serverError } = useFetch(
-    "GET",
-    `${lookUpPet}${id}`,
-    null,
-    [id],
-    tokenHeaders && tokenHeaders.headers
-  );
+  const { data, serverError } = useSWR([lookUpPet + id, tokenHeaders], fetcher);
+
+  if (serverError) console.log(serverError);
+  if (!data) console.log(data);
+
+  useEffect(() => {
+    setIsLoading(false);
+    if (!data && !serverError) {
+      setIsLoading(true);
+    } else {
+      setIsLoading(false);
+    }
+  }, [data]);
+
```

Comment:
I think `LoaderComponent` is messing things up so probably be best to remove it. I was trying to be clever haha

```
  const fetcher = (url) =>
    axios.get(url, tokenHeaders).then((res) => res.data.animal);

  const { data, serverError } = useSWR(lookUpPet + id, fetcher);
  const isLoading = !serverError && !data;
  if (serverError) return <h1>Error Loading</h1>;
  if (isLoading)
    return (
      <Spinner animation="grow" variant="primary" role="status">
        <span className="visually-hidden">Loading...</span>
      </Spinner>
    );
    ```

Reply:
Alright. I'll change that. Let's see what will happen